### PR TITLE
remove static payment parser and add test cases for parser

### PR DESCRIPTION
**Pull request by mdayican**
JIRA link (if applicable)
https://tools.hmcts.net/jira/browse/BPS-765

Change description

parser from binary azure message to paymentMessage model

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[x] No

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParserTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+public class PaymentMessageParserTest {
+
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    private PaymentMessageParser paymentMessageParser;
+
+    @BeforeEach
+    public void before() {
+        paymentMessageParser = new PaymentMessageParser(objectMapper);
```
Comment (doncem) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParserTest.java]: We can just build a static final instead

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import java.io.IOException;
+import java.util.List;
+
+@Service
+public class PaymentMessageParser {
+
+    @Autowired
+    private final ObjectMapper objectMapper;
+
+    private static final Throwable errorCause = new RuntimeException("Message Binary data is null");
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java]: ??

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParserTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+
+    private MessageBody getValidMessageBody() {
+        return fromBinaryData(ImmutableList.of(paymentMessageJsonAsByte(
+            "232131313121",
+            false
+        )));
+
+
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParserTest.java]: lets get rid off those extra empty lines

Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParserTest.java]: done

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParserTest.java (hunk 1 above)
Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParserTest.java]: done

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.exceptions.InvalidMessageException;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.service.servicebus.model.PaymentMessage;
+
+import java.io.IOException;
+import java.util.List;
+
+@Service
+public class PaymentMessageParser {
+
+    @Autowired
```
[the remark this answers is not in the document]
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java]: removed

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java (hunk 2 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java]: to throw exception as InvalidMessageException wraps Throwable.

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java]: still there

[the remark this answers is not in the document]
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java]: now removed

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/data/producer/SamplePaymentMessageData.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                )
+                .toString();
+        } catch (Exception e) {
+            throw new RuntimeException("Could not make paymentMessageJson", e);
+        }
+
+    }
+
+    public static PaymentMessage paymentMessage(String ccdCaseNumber, boolean isExceptionRecord) {
+        return new PaymentMessage(ENVELOPE_ID, ccdCaseNumber, isExceptionRecord, PO_BOX, JURSIDICTION,
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/data/producer/SamplePaymentMessageData.java]: this is not properly formatted ?

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/data/producer/SamplePaymentMessageData.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                .put("is_exception_record", isExceptionRecord)
+                .put("po_box", PO_BOX)
+                .put("jurisdiction", JURSIDICTION)
+                .put("payments", new JSONArray()
+                    .put(new JSONObject().put("document_control_number", DCN_1))
+                    .put(new JSONObject().put("document_control_number", DCN_2))
+                )
+                .toString();
+        } catch (Exception e) {
+            throw new RuntimeException("Could not make paymentMessageJson", e);
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/data/producer/SamplePaymentMessageData.java]: can just throws in method signature no need for try catch

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+            return objectMapper.readValue(getBinaryData(messageBody), PaymentMessage.class);
+        } catch (IOException exc) {
+            throw new InvalidMessageException(exc);
+        }
+    }
+
+    private static byte[] getBinaryData(MessageBody messageBody) {
+        List<byte[]> binaryData = messageBody.getBinaryData();
+        if (binaryData == null) {
+            throw new InvalidMessageException(ERROR_CAUSE);
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java]: does it need to have Throwable or can it just include string message ?

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java]: nopes not yet - import statement

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParserTest.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+        PaymentMessage expected = paymentMessage("232131313121", false);
+        PaymentMessage paymentMessage = paymentMessageParser.parse(getValidMessageBody());
+        assertThat(paymentMessage).isEqualToComparingFieldByFieldRecursively(expected);
+    }
+
+    @Test
+    public void should_throw_invalidMessageException_when_queue_message_is_invalid() {
+        assertThatThrownBy(
+            () -> paymentMessageParser.parse(fromBinaryData(ImmutableList.of("parse exception".getBytes()))))
+            .isInstanceOf(InvalidMessageException.class);
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParserTest.java]: also we can check for message -Error message

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/data/producer/SamplePaymentMessageData.java (hunk 6 above)
Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/data/producer/SamplePaymentMessageData.java]: I do not see any drawback but I changed.

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/data/producer/SamplePaymentMessageData.java (hunk 5 above)
Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/data/producer/SamplePaymentMessageData.java]: no, changed

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java (hunk 2 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java]: removed Throwable, return message

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParserTest.java (hunk 8 above)
Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParserTest.java]: added

File: src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java (hunk 7 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/payment/processor/service/servicebus/PaymentMessageParser.java]: changed

**Pull request merged**